### PR TITLE
Implement splash screen 

### LINKED ISSUE #822: Show a splash screen until the MIDlet draws to the screen
Currently we have a long startup time. It would be nice to show a screen saying something like "Please wait while your app loads" and "Powered by J2ME.js".

### PR DESCRIPTION
**Pull request by TimAbraldes**
Fixes #822.

Maybe we can get someone to choose better verbiage. This looks a little wonky on device (the title should probably wrap but it doesn't) but it's basically what I was thinking.

### CODE REVIEW

**Comment by mykmelez** [comment]
> NOP change to make travis run again

Instead of adding a space, use `git commit --allow-empty` to add a commit without making any changes.

But you don't even need to do that, since you can restart any build on Travis by pressing the *Restart Build* button in the top-right area of the build page, next to the *Download Log* button. If you don't see that button, then try signing into Travis (creating an account on the site if necessary).

File: main.html [hunk 1]
```diff
@@ -110,6 +110,18 @@
     <button id="fire">fire</button>
   </div>
 
+  <div class="splash-screen" id="splash-screen">
+    <h1 class="title">Please wait while your app loads</h1>
```
Comment (mykmelez) [inline main.html]: I would make this say *Loading [app name]…*.

File: main.html [hunk 2]
```diff
@@ -110,6 +110,18 @@
...
+    <div class="vertical-flex-container">
+      <div class="flex-buffer"></div>
+      <div>
+        <progress></progress>
+      </div>
+      <div class="flex-buffer"></div>
+    </div>
+    <p class="text">Powered by j2me.js</p>
```
Comment (mykmelez) [inline main.html]: I wouldn't display this to users, as it isn't understandable nor useful to them.

File: midp/gfx.js [hunk 3]
```diff
@@ -59,6 +59,7 @@ var currentlyFocusedTextEditor;
 
     Native.create("com/sun/midp/lcdui/DisplayDevice.gainedForeground0.(II)V", function(hardwareId, displayId) {
         console.warn("DisplayDevice.gainedForeground0.(II)V not implemented (" + hardwareId + ", " + displayId + ")");
+        document.getElementById("splash-screen").style.display = "none";
```
Comment (mykmelez) [inline midp/gfx.js]: Is there anything else to do here? If not, we should remove the "not implemented" warning.

File: style/main.css [hunk 4]
```diff
@@ -148,6 +148,71 @@ button {
...
+.autosize .splash-screen {
+  display: flex;
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+}
+
+.debug-mode .splash-screen {
```
Comment (mykmelez) [inline style/main.css]: This ruleset should go into the *STYLE OVERRIDES FOR DEBUG MODE* section.

File: midp/gfx.js (hunk 3 above)
Comment (TimAbraldes) [inline midp/gfx.js]: I believe we need to modify this function to fix #823, but that doesn't necessarily mean that this warning is useful to us

**Pull request merged**